tests/t319-basis.c

```diff
@@ -82,66 +148,30 @@ int main(int argc, char **argv) {
     CeedBasisCreateTensorH1Lagrange(ceed, dim, 1, p_to, q, CEED_GAUSS, &basis_to);
     CeedBasisCreateProjection(basis_from, basis_to, &basis_project);
 
-    // Setup coarse solution
-    {
-      const CeedScalar *x_array;
-      CeedScalar        u_array[p_from_dim];
-
-      CeedVectorGetArrayRead(x_from, CEED_MEM_HOST, &x_array);
-      for (CeedInt i = 0; i < p_from_dim; i++) {
-        CeedScalar coord[dim];
-        for (CeedInt d = 0; d < dim; d++) coord[d] = x_array[p_from_dim * d + i];
-        u_array[i] = Eval(dim, coord);
-      }
-      CeedVectorRestoreArrayRead(x_from, &x_array);
-      CeedVectorSetArray(u_from, CEED_MEM_HOST, CEED_COPY_VALUES, u_array);
-    }
+    VerifyProjectedBasis(basis_project, dim, p_to_dim, p_from_dim, x_to, x_from, u_to, u_from, du_to);
 
-    // Project to fine basis
-    CeedBasisApply(basis_project, 1, CEED_NOTRANSPOSE, CEED_EVAL_INTERP, u_from, u_to);
-
-    // Check solution
+    // Test projection on mixed-tensor bases
+    CeedBasis basis_from_nontensor, basis_project_mixed;
     {
-      const CeedScalar *x_array, *u_array;
-
-      CeedVectorGetArrayRead(x_to, CEED_MEM_HOST, &x_array);
-      CeedVectorGetArrayRead(u_to, CEED_MEM_HOST, &u_array);
-      for (CeedInt i = 0; i < p_to_dim; i++) {
-        CeedScalar coord[dim];
-        for (CeedInt d = 0; d < dim; d++) coord[d] = x_array[d * p_to_dim + i];
-        const CeedScalar u = Eval(dim, coord);
-        if (fabs(u - u_array[i]) > tol) printf("[%" CeedInt_FMT ", %" CeedInt_FMT "] %f != %f\n", dim, i, u_array[i], u);
-      }
-      CeedVectorRestoreArrayRead(x_to, &x_array);
-      CeedVectorRestoreArrayRead(u_to, &u_array);
+      CeedElemTopology  topo;
+      CeedInt           num_comp, num_nodes, nqpts;
+      const CeedScalar *interp, *grad, *q_ref, *q_weights;
+
+      CeedBasisGetTopology(basis_from, &topo);
+      CeedBasisGetNumComponents(basis_from, &num_comp);
+      CeedBasisGetNumNodes(basis_from, &num_nodes);
+      CeedBasisGetNumQuadraturePoints(basis_from, &nqpts);
+      CeedBasisGetInterp(basis_from, &interp);
+      CeedBasisGetGrad(basis_from, &grad);
+      CeedBasisGetQRef(basis_from, &q_ref);
+      CeedBasisGetQWeights(basis_from, &q_weights);
```

Comment:
So `CeedBasisInterp` has a tensor-product 1D version for it, but `CeedBasisQref` and `CeedBasisGetQWeights`. Is there any reason why we don't have full vs 1D versions of the quadrature reference data?

Reply:
Is there a reason? No.

If I'm remembering correctly those are BS for the new projection Basis and we should be using [1...] for the weights and [0...] for the pts

Reply:
I was about to go ahead and implement a `CeedBasisQRef1D` and `CeedBasisQWeights1D`, but just realized that would actually break the current interface for `CeedBasisQRef` and `CeedBasisQWeights`. Like, if you call those for tensor basis now, you'll get the 1D representation. But the new version would change it's output. I don't like breaking code, but this seems like it was almost broken to begin with (or at least misleading).

Reply:
Yeah, I'd leave this alone for now, at least until we can discuss and see what we want to do

But I think updating the new q_ref and q_weight to not be lies would be great

Reply:
Yeah, I can correct the projection basis. 

I'll made something in Zulip to discuss. It just makes the test case more annoying to setup. haha

Reply:
Oh, it's q_ref and q_weights are optional inputs to `CeedBasisCreateH1`. So not a pain at all.

Reply:
Looks like `q_ref` and `q_weight` is already being set to zero (via calloc). Do we want to change the behavior there?

I feel like setting them be `NULL` would also make sense.

Reply:
Yeah, since they are optional inputs for `CeedBasisCreateH1` and `CeedBasisCreateTensorH1` we really should use `NULL` there since those aren't valid for this Basis

Reply:
Done.